### PR TITLE
feat(v2): allow creation of attachment fields in admin form builder

### LINKED ISSUE #3400: [React/Builder] Implement Attachment field drawer and preview

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR adds ability to create attachment fields in admin form builder screen. 

Closes #3400 

## Solution

**Features**:

- allow creation of attachment fields in admin form builder screen

## Before & After Screenshots
New stories added for EditAttachment component.

### CODE REVIEW

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            value: AttachmentSize.SevenMb,
+            label: `${AttachmentSize.SevenMb} MB`,
+          },
+        ]
+      case FormResponseMode.Encrypt:
+        return Object.values(AttachmentSize).map((size) => ({
+          value: size,
+          label: `${size} MB`,
+        }))
+    }
```
Comment (timotheeg) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx:129]: Not sure if Typescript will like it, but I'd share some of the formatting code here like this:

```suggestion
    let sizes;
    
    if (form.responseMode === FormResponseMode.Email) {
        sizes = [
          AttachmentSize.OneMb,
          AttachmentSize.TwoMb,
          AttachmentSize.ThreeMb,
          AttachmentSize.FourMb,
          AttachmentSize.SevenMb
        ]
    } else if (form.responseMode === FormResponseMode.Encrypt) {
      sizes = Object.values(AttachmentSize);
    }
    
    return sizes.map((size) => ({
        value: size,
        label: `${size} MB`,
      }))
```

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      .reduce((sum, ff) => sum + Number(ff.attachmentSize), 0)
+  }, [field._id, form?.form_fields])
+
+  const maxTotalSizeMb: number = useMemo(() => {
+    if (!form?.responseMode) return 0
+    switch (form.responseMode) {
+      case FormResponseMode.Email:
+        return 7
+      case FormResponseMode.Encrypt:
+        return 20
```
Comment (timotheeg) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx:100]: Should we not use the constants here? 🤔 
```suggestion
        return Number(AttachmentSize.SevenMb)
      case FormResponseMode.Encrypt:
        return Number(AttachmentSize.TwentyMb)
```

**Review by timotheeg — COMMENTED** [review]
lgtm, only have some minor questions

**Review by timotheeg — APPROVED** [review]

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx]: 849f756

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx]: d869055

**Review by timotheeg — APPROVED** [review]

**Pull request merged**